File: Java/src/main/java/QuestionBank/Solution701_800/Solution744.java
package QuestionBank.Solution701_800;

/**
 * @author windows
 */
public class Solution744 {
    public char nextGreatestLetter(char[] letters, char target) {
        for (int i = 0; i < letters.length; i++) {
            if (letters[i] > target) {
                return letters[i];
            }
        }
        return letters[0];
    }


    public static void main(String[] args) {
        Solution744 solution744 = new Solution744();
    }
}

File: C++/src/Solution701_800/Solution744.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution744.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

char Solution::nextGreatestLetter(vector<char> &letters, char target) {
    int n = letters.size();
    int left = -1, right = n;
    while (left != right - 1) {
        int mid = (left + right) / 2;
        if (letters[mid] <= target) {
            left = mid;
        } else {
            right = mid;
        }
    }
    return right == n ? letters[0] : letters[right];
}
